src/main/java/com/mg105/use_cases/MapGenerator.java

```diff
@@ -0,0 +1,58 @@
+package com.mg105.use_cases;
+
+import com.mg105.entities.*;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MapGenerator {
+    private final GameState state;
+
+    /**
+     * Creates a new MapGenerator that acts on a GameState.
+     *
+     * state must not be null.
+     *
+     * @param state the game state.
+     */
+    public MapGenerator(GameState state) {
+        this.state = state;
+    }
+
```

Comment:
no sure whether the MapGenerator class needs a variable that contains GameState as an instance variable, but still good coding.